lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -112,12 +112,20 @@ def add_files_to_build_phases
               next unless target.requires_frameworks?
 
               resource_refs = file_accessor.resources.flatten.map do |res|
-                project.reference_for_path(res)
+                ref = project.reference_for_path(res)
+
+                # Some nested files are not directly present in the Xcode project, such as the contents
+                # of an .xcdatamodeld directory. These files are implicitly included by including their
+                # parent directory.
+                next if ref.nil?
+
+                # For variant groups, the variant group itself is added, not its members.
+                next ref.parent if ref.parent.is_a?(Xcodeproj::Project::Object::PBXVariantGroup)
```

Comment:
Since we're basically copying this code from down below, maybe we could wrap it into a nice method on self and reuse it here?